Source/NameServer/AddressBook/AddressBook_Record.cpp

```diff
@@ -0,0 +1,49 @@
+#include "AddressBook_Record.hpp"
+
+namespace AddressBook
+{
+
+/*==============================================================================
+ * Record_t::Record_t(): Default initialiser for a Record_t.
+ *============================================================================*/
+Record_t::Record_t()
+{
+    Attribute = -1;     // Set to -1 to ensure that we don't go out of range.
```

Comment:
Ominous

Reply:
Because it is an index and attributes are optional, putting in 0 is immediately potentially out of range. -1 indicates that there are NO attributes